python/nav/ipdevpoll/plugins/lldp.py

```diff
@@ -137,3 +167,90 @@ def _store_unidentified(self, record):
         neighbor.remote_id = record.chassis_id
         neighbor.remote_name = record.sysname
         neighbor.source = SOURCE
+
+
+class LLDPNeighbor(Neighbor):
+    "Parses an LLDP tuple from nav.mibs.lldp_mib to identify a neighbor"
+
+    def _identify_netbox(self):
+        chassid = self.record.chassis_id
+        netbox = None
+        if chassid:
+            lookup = None
+            if isinstance(chassid, lldp_mib.IdSubtypes.macAddress):
+                lookup = self._netbox_from_mac
+            elif isinstance(chassid, lldp_mib.IdSubtypes.networkAddress):
+                lookup = self._netbox_from_ip
+            elif isinstance(chassid, lldp_mib.IdSubtypes.local):
+                lookup = self._netbox_from_local
+
+            if lookup:
+                netbox = lookup(str(chassid))
+
+        if not netbox and self.record.sysname:
+            netbox = self._netbox_from_sysname(self.record.sysname)
+
+        return netbox
+
+    def _netbox_from_local(self, chassid):
+        netbox = self._netbox_query(info__key=INFO_KEY_LLDP_INFO,
```

Comment:
`_netbox_query` does not take keyword arguments, so this fails miserably. 

```pytb
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/twisted/internet/defer.py", line 500, in errback
    self._startRunCallbacks(fail)
  File "/usr/local/lib/python3.5/dist-packages/twisted/internet/defer.py", line 567, in _startRunCallbacks
    self._runCallbacks()
  File "/usr/local/lib/python3.5/dist-packages/twisted/internet/defer.py", line 653, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
  File "/usr/local/lib/python3.5/dist-packages/twisted/internet/defer.py", line 1442, in gotResult
    _inlineCallbacks(r, g, deferred)
--- <exception caught here> ---
  File "/usr/local/lib/python3.5/dist-packages/twisted/internet/defer.py", line 1384, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/usr/local/lib/python3.5/dist-packages/twisted/python/failure.py", line 408, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/source/python/nav/ipdevpoll/plugins/lldp.py", line 71, in handle
    yield run_in_thread(self._process_remote)
  File "/usr/local/lib/python3.5/dist-packages/twisted/python/threadpool.py", line 250, in inContext
    result = inContext.theWork()
  File "/usr/local/lib/python3.5/dist-packages/twisted/python/threadpool.py", line 266, in <lambda>
    inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
  File "/usr/local/lib/python3.5/dist-packages/twisted/python/context.py", line 122, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/usr/local/lib/python3.5/dist-packages/twisted/python/context.py", line 85, in callWithContext
    return func(*args,**kw)
  File "/source/python/nav/ipdevpoll/db.py", line 100, in _reset
    return func(*args, **kwargs)
  File "/source/python/nav/ipdevpoll/plugins/lldp.py", line 116, in _process_remote
    neighbors = [LLDPNeighbor(lldp) for lldp in self.remote]
  File "/source/python/nav/ipdevpoll/plugins/lldp.py", line 116, in <listcomp>
    neighbors = [LLDPNeighbor(lldp) for lldp in self.remote]
  File "/source/python/nav/ipdevpoll/neighbor.py", line 117, in __init__
    self.identify()
  File "/source/python/nav/ipdevpoll/neighbor.py", line 120, in identify
    self.netbox = self._identify_netbox()
  File "/source/python/nav/ipdevpoll/plugins/lldp.py", line 188, in _identify_netbox
    netbox = lookup(str(chassid))
  File "/source/python/nav/ipdevpoll/plugins/lldp.py", line 198, in _netbox_from_local
    value=chassid).first()
builtins.TypeError: _netbox_query() got an unexpected keyword argument 'info__key'

```

1. You should express your filter as a `Q` object and submit that as the first argument instead.
2. Also, the function actually returns a Netbox object, not a QuerySet, so the call to `.first()` will also not work.
3. Some devices *do* actually report a sysname string when the subtype of the value is `local`, so that should still be the fallback.